app/Knights.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations  # Enable forward annotations
+
+from typing import List, Optional
+
+from app.armour import Armour
+from app.weapon import Weapon
+from app.potion import Potion
+
+
+class Knight:
+    def __init__(
+        self,
+        name: str,
+        power: int,
+        hp: int,
+        armour: Optional[List[dict]] = None,
+        weapon: Optional[dict] = None,
+        potion: Optional[dict] = None,
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = 0
+
+    def potion_effect(self) -> None:
+        for effect, value in self.potion.effect.items():
+            if effect == "hp":
+                self.hp += value
+            elif effect == "protection":
+                self.protection += value
+            elif effect == "power":
+                self.power += value
+
+    def add_weapon(self, weapon: dict) -> None:
+        self.weapon = Weapon(weapon["name"], weapon["power"])
+
+    def apply_weapon(
+        self,
+    ) -> None:
+        self.power += self.weapon.power
+
+    def add_armour(self, armour: dict) -> None:
+        if armour:
+            self.armour = []
+            for fighter in armour:
+                self.armour.append(
+                    Armour(
+                        fighter["part"],
+                        fighter["protection"]
+                    )
+                )
+
+    def apply_armour(
+        self,
+    ) -> None:
+        if self.armour:
```

Comment:
If self.armour is  an empty list, you can omit this condition because loop will be exited automatically